packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -51,10 +51,11 @@ import {
     IFileEntry,
     snapshotExpirySummarizerOps,
```

Comment:
OdspDocumentStorageService.getVersions (line 357), where we fetch snapshot via this.cache.persistedCache.get - we need to both store epoch, and retrieve epoch from cache, and populate with it Fetcher object.
Also such population should check if epoch is already known and fail if they differ, as it's possible we fetch snapshot after doing some other operation on file that retrieved epoch.

I think this actually applies to any operation with cache, current of future (for example in future we may add ability to store blobs & ops). I think actually right design & behavior is to have an adaptive layer over cache, such that it talks to Fetcher object and if cache has mismatched epoch, then cache entry is deleted and not used, but we allow to keep going. And if cache is used to get first info about file (like snapshot), then epoch is populated from cache.

Reply:
Added logic of adding epoch to cache values. Then on fetch from cache, I am now checking the epoch value and then either initializing the fetcher epoch value from cached value or if match then clear all values from cache.